src/Hydrator/EloquentHydrator.php

```diff
@@ -253,11 +254,16 @@ protected function deserializeDate($value)
     /**
      * Is this resource key a date attribute?
      *
+     * @param $record
      * @param $resourceKey
      * @return bool
      */
-    protected function isDateAttribute($resourceKey)
+    protected function isDateAttribute($record, $resourceKey)
     {
+        if(is_null($this->dates))
+        {
+            return in_array(Str::snake($resourceKey), $record->getDates(), true);
+        }
         return in_array($resourceKey, $this->dates, true);
```

Comment:
Just one line-break before the return statement, then I'm done OCD'ing all over your code. 😅